src/pruna/config/target_modules.py

```diff
@@ -0,0 +1,211 @@
+        The name of the hyperparameter.
+    default_value : Optional[TARGET_MODULES_TYPE]
+        The default value of the hyperparameter.
+    meta : Any
+        Meta data describing the hyperparameter.
+    """
+
+    def __init__(self, name: str, default_value: Optional[TARGET_MODULES_TYPE] = None, meta: Any = None) -> None:
+        super().__init__(name, default_value, meta=meta)
+
+    @override
+    def legal_value(self, value: TARGET_MODULES_TYPE | None):  # type: ignore[override]  # numpydoc ignore=GL08
+        """
+        Check if a value is a valid target modules of type TARGET_MODULES_TYPE.
+
+        Parameters
+        ----------
+        value : Any
+            The value to check.
+
+        Returns
+        -------
+        bool or numpy.ndarray
+            `True` if the value is of type TARGET_MODULES_TYPE, `False` otherwise.
+        """
+        # ensure the value is a TARGET_MODULES_TYPE to make errors more explicit for the user
+        if value is None:
+            pass
+        elif not isinstance(value, dict):
+            raise TypeError(f"Target modules must be a dictionary with keys 'include' and/or 'exclude'. Got: {value}")
+        elif any(key not in ["include", "exclude"] for key in value):
+            raise ValueError(f"Target modules must only use keys 'include' and/or 'exclude'. Got: {list(value.keys())}")
+        elif any(not isinstance(patterns, list) for patterns in value.values()):
+            raise TypeError(
+                f"Target modules must be a dictionary with lists of fnmatch patterns as values. Got: {value}"
+            )
+        else:
+            all_patterns = [pattern for patterns in value.values() for pattern in patterns]
+            unrecognized_patterns = [pattern for pattern in all_patterns if not isinstance(pattern, str)]
+            if unrecognized_patterns:
+                raise TypeError(
+                    "Target modules must be a dictionary with lists of "
+                    "Unix shell-style wildcards (fnmatch-style) patterns as values. "
+                    f"Could not recognize the following as fnmatch patterns: {unrecognized_patterns}."
+                )
+
+        # handle default value: modify the dict in place to have a match between the value and default value
+        if value is None:
+            pass  # chosing a default value is left to the algorithm based on the model
+        elif "include" not in value:
+            value["include"] = ["*"]
+        elif "exclude" not in value:
+            value["exclude"] = []  # for consistency
+
+        return super().legal_value(value)
+
+
+def expand_list_of_modules_paths(target_modules: TARGET_MODULES_TYPE, model: Any) -> List[str]:
+    """
+    Convert the target modules to a list of module paths.
+
+    Parameters
+    ----------
+    model : Any
+        The model to get the module paths from.
+    target_modules : TARGET_MODULES_TYPE
+        The target modules to convert to a list of module paths.
```

Comment:
switch order to match the signature
```suggestion
    Parameters
    ----------
    target_modules : TARGET_MODULES_TYPE
        The target modules to convert to a list of module paths.
    model : Any
        The model to get the module paths from.
```